cocos/core/pipeline/render-additive-light-queue.ts

```diff
@@ -349,11 +319,15 @@ export class RenderAdditiveLightQueue {
         const isSupportHalfFloat = supportsHalfFloatTexture(device);
         const linear = (shadowInfo.linear && isSupportHalfFloat) ? 1.0 : 0.0;
         const packing = shadowInfo.packing ? 1.0 : (isSupportHalfFloat ? 0.0 : 1.0);
+        const multiLightDSManager = this._pipeline.multiLightDSManager;
 
         for (let i = 0; i < this._validLights.length; i++) {
             const light = this._validLights[i];
-            const descriptorSet = this._getOrCreateDescriptorSet(light);
+            const descriptorSet = multiLightDSManager.getOrCreateDescriptorSet(light);
+            multiLightDSManager.updateDescriptorSet(light);
```

Comment:
I don't see why this has to be done every frame. You can always bind all the existing bindings to the newly created DS, and just use it afterwards. The updates by other systems (skybox, etc.) later should invoke the GlobalDSManager.update method themselves.

```js
class GlobalDSManager {
    private _globalDescriptorSets: Map<number, DescriptorSet>; 

    public bindBuffer(binding, buffer);
    public bindSampler(binding, sampler);
    public bindTexture(binding, texture);
    public update();

    public getOrCreateInstance(idx: number); // if we are creating a new one, make a copy of all the existing bindings
}
````

Reply:
Like skybox is bound internally in its own class，I don't have access to the.

There is also a question of update sequence here. For example, if the IBL is added first and the light is updated later, the lightDescriptorSet created later will have no way to know that there is a binding to the cc_ENVIRONMENT location

Reply:
using like it